File: solutions/1480. Running Sum of 1d Array.cpp
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        vector<int> sum;
        int t=0;
        for(int i=0;i<nums.size();i++){
            sum.push_back(t+nums[i]);
            t=t+nums[i];
        }
        return sum;
    }
};

File: solutions/1480. Running Sum of 1d Array.py
class Solution:
    #def runningSum(self, nums: List[int]) -> List[int]:
    #    s=0,ans=[]
    #    for i in nums:
    #        s+=i
    #        ans.append(s)
    #    return ans
​
    #to do in O(1) space
    def runningSum(self, nums: List[int]) -> List[int]:
        for i in range(1,len(nums)):
            nums[i]+=nums[i-1]
        return nums
